### PR TITLE
Drop then recreate objects on bootstrap (fixes some comments form #3489)

### PR DESCRIPTION
**Pull request by bilboquet**
* [x] document all added functions
* [x] try in sandbox /simulation/labnet
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

File: massa-bootstrap/src/client.rs [hunk 1]
```diff
@@ -168,8 +168,13 @@ async fn stream_final_state_and_consensus(
                         last_ops_step: StreamingStep::Started,
                         last_consensus_step: StreamingStep::Started,
                     };
+
+                    // TODO:
+                    // Building a new FinalState outside of this function would probably be better.
+                    // There are to much deps to manage for the moment.
```
Comment (sydhds) [inline massa-bootstrap/src/client.rs:174]: to much -> too much?

File: massa-pos-exports/src/pos_final_state.rs [hunk 2]
```diff
@@ -64,12 +64,22 @@ impl PoSFinalState {
...
-    /// Reset the state of the PoS final state
-    ///
-    /// USED ONLY FOR BOOTSTRAP
-    pub fn reset(&mut self) {
-        self.cycle_history.clear();
-        self.deferred_credits = DeferredCredits::default();
+    /// Inspired by mem::take()
+    /// This function reinitialize in place the given object.
+    /// It reuses what it can from the given reference.
+    /// The implementation ensure that all fields are tracked.
```
[the remark this answers is not in the document]
Comment (bilboquet) [reply massa-pos-exports/src/pos_final_state.rs]: done

File: massa-final-state/src/final_state.rs [hunk 3]
```diff
@@ -86,9 +86,9 @@ impl FinalState {
     pub fn reset(&mut self) {
```
Comment (bilboquet) [inline massa-final-state/src/final_state.rs:86]: Tradeoff here see the followup in  #3728

Comment (damip) [reply massa-final-state/src/final_state.rs:86]: Why not just recreate the final_state object everytime we attempt a bootstrap ?

Comment (bilboquet) [reply massa-final-state/src/final_state.rs:86]: IIRC I did not find a way with minimal modifications mainly because the final_state is behind a & mut.

Comment (AurelienFT) [reply massa-final-state/src/final_state.rs:86]: After analyzing the code. The final state is created before the bootstrap and sent as a `RwLock` to the bootstrap but I don't think it's necessary as it's not used anywhere before getting filled by the bootstrap. So we could create the first one before the bootstrap or inside, and return the filled one to be used in the rest of the code.

As the code that launch the bootstrap is under a `tokio::select` I think it should be better to continue on that PR after bootstrap de-async

**Pull request closed without merging**